core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -321,7 +322,7 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
       // scan for modified applications, replay and merge them
       val logInfos: Seq[FileStatus] = statusList
         .filter { entry =>
-          val prevFileSize = fileToAppInfo.get(entry.getPath()).map{_.fileSize}.getOrElse(0L)
+          val prevFileSize = fileToAppInfo.asScala.get(entry.getPath()).map{_.fileSize}.getOrElse(0L)
```

Comment:
`asScala` produces a wrapper, right? in which case this can be called once, at declaration above. If it produces a clone, this doesn't work anyway!
But why is this needed anyway? you can `get` elements from the Java map already.